kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -777,24 +778,33 @@ internal class CodeWriter(
       simpleName: String,
       canonicalNamesToQualifiedNames: Map<String, T>,
       capitalizeAliases: Boolean,
+      imported: Set<String>,
     ): List<Pair<String, T>> {
       val canonicalNameSegmentsToQualifiedNames = canonicalNamesToQualifiedNames.mapKeys { (canonicalName, _) ->
         canonicalName.split('.')
           .dropLast(1) // Last segment of the canonical name is the simple name, drop it to avoid repetition.
           .filter { it != "Companion" }
           .map { it.replaceFirstChar(Char::uppercaseChar) }
       }
+      val minimumNumberOfSegments = canonicalNameSegmentsToQualifiedNames.keys.minOf { it.size }
       val aliasNames = mutableMapOf<String, T>()
       var segmentsToUse = 0
       // Iterate until we have unique aliases for all names.
-      while (aliasNames.size != canonicalNamesToQualifiedNames.size) {
+      outer@ while (aliasNames.size != canonicalNamesToQualifiedNames.size) {
         segmentsToUse += 1
         aliasNames.clear()
         for ((segments, qualifiedName) in canonicalNameSegmentsToQualifiedNames) {
           val aliasPrefix = segments.takeLast(min(segmentsToUse, segments.size))
             .joinToString(separator = "")
             .replaceFirstChar { if (!capitalizeAliases) it.lowercaseChar() else it }
-          val aliasName = aliasPrefix + simpleName.replaceFirstChar(Char::uppercaseChar)
+          val aliasSuffix = (0..<(segmentsToUse - segments.size)).joinToString("") { "_" }
```

Comment:
I think this is
```suggestion
          "_".repeat(segmentsToUse - segments.size)
```

Reply:
Ah perfect, thanks!

Reply:
Because this value can (and usually is) negative, this needed to be `"_".repeat((segmentsToUse - segments.size).coerceAtLeast(0))`